### DISCUSSION TITLE
IPv6 on AWS EC2 - Debian 11

### DISCUSSION THREAD

**Discussion by Squiggly9461** [angristan/wireguard-install#422] [Help]
I just noticed when I install either wireguard or OpenVPN created by angristan the IPv6 is no longer reachable. 

IPv6 has been properly configured on VPC and is reachable on the internet. I have a Vultr VPS as well and its working fine. 

Any idea or are there any changes should we made for AWS EC2 instances? This also affects AWS Lightsail.

**Comment by angristan** [comment]
Can you precise:

> the IPv6 is no longer reachable

Which IPv6, and from where?

**Reply by MSe1969** [reply]
Wireguard requires IP-Forwarding, which is therefore enabled by the 'wireguard-install' script.

Your issue is indeed a VPS-provider specific issue:
On Debian (Ubuntu seems different here), enabling ip6 forwarding disables the so-called "router advertisement", you can see that in the comments section of **/etc/sysctl.conf**:
```
# Uncomment the next line to enable packet forwarding for IPv6
#  Enabling this option disables Stateless Address Autoconfiguration
#  based on Router Advertisements for this host
#net.ipv6.conf.all.forwarding=1
```
So if your provider uses router advertisement as part of the IPv6 autoconfiguration, you get - at least on a Debian system - an issue with IPv6, when you enable IPv6 forwarding.

On an AWS Lightsail instance with Debian, the problem is solved as follows:
First, get the interface name (e.g. `ip route show default`) - it returns something like "default via [ip-address] dev [interface] [further stuff], so you take the expression after "dev" - should normally be **eth0** on a Lightsail instance.

Next add the entry "net.ipv6.conf.**eth0**.accept_ra=2" into /etc/sysctl.conf (adapt accordingly with the real interface used) : 
```
# Uncomment the next line to enable packet forwarding for IPv6
#  Enabling this option disables Stateless Address Autoconfiguration
#  based on Router Advertisements for this host
net.ipv6.conf.all.forwarding=1
net.ipv6.conf.eth0.accept_ra=2
```
(If your IPv6 is already screwed, reboot the instance afterwards `sudo shutdown -r`)

**Reply by nameup2** [reply]
Thanks bro,you solved my problem!this problem has troubled me for a month
